src/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs

```diff
@@ -1650,4 +1649,13 @@ public sealed class SocketsHttpHandlerTest_Cookies_Http2 : HttpClientHandlerTest
         protected override bool UseSocketsHttpHandler => true;
         protected override bool UseHttp2LoopbackServer => true;
     }
+
+    [ConditionalClass(typeof(PlatformDetection), nameof(PlatformDetection.SupportsAlpn))]
+    public sealed class SocketsHttpHandlerTest_HttpClientHandlerTest_Http2 : HttpClientHandlerTest
```

Comment:
This means that every single test in HttpClientHanderTest is going to run in HTTP2 loopback mode. The problem is that most of these tests are not generic loopback based tests, so when they run in "HTTP2 mode" they are really just running in HTTP/1.1 mode again. That seems bad.

I believe we need to separate the generic loopback tests into a separate class so that we can specifically run just those tests in HTTP2 mode (as well as HTTP/1.1 of course).

Reply:
my assumption is that we would fix them over time and most of the tests could run for both HTTP/1.x and 2.0. Looking at the file HTTP/1.x tests seems minority and it may be easier to move them instead to HTTP/1 specific file.